repos/jsutils/src/array/areSetEqual.js

```diff
@@ -0,0 +1,36 @@
+/** @module array */
+
+import { validate } from '../validation'
+import { isArr } from './isArr'
+
+/**
+ * Checks if arrays are set-equal: they contain the same elements,
+ * but element frequencies don't matter.
+ * Does this with one pass over each array and an auxilliary set.
+ * @function
+ * @param {Array<*>} arr 
+ * @param {Array<*>} otherArr 
+ */
+export const areSetEqual = (arr, otherArr) => {
+  const [ valid ] = validate({ arr, otherArr }, { $default: isArr })
+  if (!valid) return null
+
+  if (arr === otherArr) return true
+
+  const longest = arr.length > otherArr.length
```

Comment:
Instead of doing this compare twice, you can just do this =>
```js
  const [longest, shortest] = arr.length > otherArr.length
    ? [arr, otherArr]
    : [otherArr, arr]
```

Reply:
updated